### PR TITLE
SETI-5977: Migrating from Travis-CI to Github Action

### PR DESCRIPTION
**Pull request by helobinvn**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Set branch name and PR number
+        id: refs
+        env:
+          BRANCH_NAME_OR_REF: ${{ github.head_ref || github.ref }}
+        run: |
+          echo "::set-output name=branch_name::${BRANCH_NAME_OR_REF#refs/heads/}"
+          echo "::set-output name=pr_number::$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")"
+      - name: Coveralls Maven plugin
```
Comment (liry) [inline .github/workflows/build.yml:27]: This does not only run "Coveralls Maven plugin". It runs complete build with all the checks, compilation and tests + generates javadocs + generates and sends coverage report using that mentioned plugin

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  push:
+    branches:
+      - master
+      - /\d\.0\.0-RC/
+  pull_request:
+    branches:
+      - master
+      - /\d\.0\.0-RC/
+jobs:
+  check-coverage:
```
Comment (liry) [inline .github/workflows/build.yml:12]: The same as below - coverage is only a subset of this job

File: README.md [hunk 3]
```diff
@@ -1,11 +1,12 @@
 # GoodData Java SDK
-[![Build Status](https://travis-ci.org/gooddata/gooddata-java.png?branch=master)](https://travis-ci.org/gooddata/gooddata-java)
+![Build Status](https://github.com/gooddata/gooddata-java/actions/workflows/build.yml/badge.svg?branch=master)
```
Comment (liry) [inline README.md:2]: I would wrap it with link to GH Actions the same way as other batches.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    steps:
```
Comment (liry) [inline .github/workflows/build.yml:14]: I miss dependencies cache - see https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#caching-dependencies

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by liry** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/230/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-java-fossa-scanning-zuul-el7/69/ : FAILURE in 2m 08s (non-voting)

**Pull request merged**